pkg/kessel/config/config.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"crypto/tls"
+	"time"
+)
+
+type Config struct {
+	// Endpoint specifies the server address (host:port)
+	Endpoint string `json:"endpoint" env:"KESSEL_ENDPOINT"`
+
+	// Timeout specifies the default request timeout
+	Timeout time.Duration `json:"timeout" env:"KESSEL_TIMEOUT" default:"10s"`
+
+	// UserAgent specifies the client user agent
+	UserAgent string `json:"user_agent" env:"KESSEL_USER_AGENT" default:"kessel-go-sdk"`
+
+	// UseHTTP forces HTTP instead of gRPC
+	UseHTTP bool `json:"use_http" env:"KESSEL_USE_HTTP"`
+
+	// Insecure disables transport security
+	Insecure bool `json:"insecure" env:"KESSEL_INSECURE"`
+
+	// TLSConfig specifies custom TLS configuration
+	TLSConfig *tls.Config `json:"-"`
+
+	EnableOIDCAuth bool `json:"enableOIDCAuth" env:"KESSEL_ENABLE_OAUTH"`
+	// OAuthConfig specifies OAuth2 configuration
+	ClientID     string   `json:"client_id" env:"KESSEL_OAUTH_CLIENT_ID"`
+	ClientSecret string   `json:"client_secret" env:"KESSEL_OAUTH_CLIENT_SECRET"`
+	TokenURL     string   `json:"token_url" env:"KESSEL_OAUTH_TOKEN_URL"`
+	Scopes       []string `json:"scopes" env:"KESSEL_OAUTH_SCOPES" envSeparator:","`
```

Comment:
Re `enableOIDCAuth`, technically it is not OIDC auth, it is just OAuth2. No identity token is involved, just an access token. It's the access token that should be sent to Kessel, not the identity token. It's also kinda inconsistent with the environment variable names.

Similarly, here maybe we want a nested structure for the JSON and struct fields as well, so we can namespace things like ClientID and ClientSecret within the "oauth2" name so as to avoid collisions.